engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/EmailValidator.java

```diff
@@ -104,17 +105,26 @@ private boolean matchPart(String part, Pattern pattern, int maxLength) {
 	private String toAscii(String unicodeString) throws IllegalArgumentException {
 		String asciiString = "";
 		int start = 0;
-		int end = unicodeString.length() <= 63 ? unicodeString.length() : 63;
+		int index = getBreakingIndex( unicodeString, ASCII_MAX_LABEL_LENGTH );
+		int end = unicodeString.length() <=  index ? unicodeString.length() : index;
 		while ( true ) {
 			// IDN.toASCII only supports a max "label" length of 63 characters. Need to chunk the input in these sizes
 			asciiString += IDN.toASCII( unicodeString.substring( start, end ) );
 			if ( end == unicodeString.length() ) {
 				break;
 			}
 			start = end;
-			end = start + 63 > unicodeString.length() ? unicodeString.length() : start + 63;
+			index = getBreakingIndex( unicodeString, start + ASCII_MAX_LABEL_LENGTH );
+			end = index > unicodeString.length() ? unicodeString.length() : index;
 		}
 
 		return asciiString;
 	}
+
+	private int getBreakingIndex(String unicodeString, int index) {
+		if (unicodeString.length() <= index) {
+			return index;
+		}
+		return unicodeString.charAt( index ) == '.' ? Math.max( 0, index - 1 ) : index;
```

Comment:
@gsmet 
actually the issue is when the string passed starts with the dot. As in the test case for this issue the domain part of the email was split in two parts - first part contained a series of 'a' chars and second part was just '.com' and when this '.com' was passed to `IDN.toASCII` it was throwing an exception. 
So what I've done to overcome the problem - is to check if we are not trying to split  a string in such "wrong" way and at least leave one character before the '.' That's why I return `index - 1` in such case (as substring second parameter index is exclusive we will leave at least one character before '.')

hope I've explained it well :)